README.rst

```diff
@@ -918,13 +921,28 @@ variable.  The default is 0.5, which can be decreased to reduce RAM usage
 in memory constrained enviroments, or increased if performance starts to
 degrade.
 
+However, degraded performance due to a low cache factor, common on
+machines with slow disks, often leads to explosions in memory use due to
+requests backing up. In this case, reducing the cache factor will make
+things worse. Instead, try increasing it drastically. 2.0 is a good
+starting value.
+
 Using `libjemalloc <http://jemalloc.net/>`_ can also yield a significant
-improvement in overall amount, and especially in terms of giving back RAM
-to the OS. To use it, the library must simply be put in the LD_PRELOAD
-environment variable when launching Synapse. On Debian, this can be done
-by installing the ``libjemalloc1`` package and adding this line to
-``/etc/default/matrix-synapse``::
+improvement in overall memory use, and especially in terms of giving back
+RAM to the OS. To use it, the library must simply be put in the
+LD_PRELOAD environment variable when launching Synapse. On Debian, this
+can be done by installing the ``libjemalloc1`` package and adding this
+line to ``/etc/default/matrix-synapse``::
 
     LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libjemalloc.so.1
 
+If you're encountering high CPU use by the Synapse process itself, you
+may be encountering a bug with presence tracking that leads to a massive
+excess of outgoing federation requests (see `discussion
+<https://github.com/matrix-org/synapse/issues/3971>`_). If your server is
+also issuing far more outgoing federation requests than can be accounted
+for by your users' activity, this is a likely cause. The misbehavior can
```

Comment:
I've been encountering a somewhat steady stream of people having this problem, and who knows how many go unseen. The situation right now is pretty dire and it is difficult for an affected server operator to find this information if it's not published somewhere central.

A better solution than mentioning the README would be to disable presence by default until it's no longer a performance disaster. Is that on the table?

Reply:
I'd argue that since the title of the section is "Help!! Synapse eats all my RAM!", it is already somewhat baked in to the README except without all the pertinent information. Not everyone knows to look for that issue.

I think it is important to get this section included in Synapse 1.0 until more performance improvements come in the future.